src/aleph/netty.clj

```diff
@@ -203,41 +201,30 @@
   (->> (bs/convert x (bs/stream-of ByteBuf) {:chunk-size chunk-size})
     (s/onto nil)))
 
+(defn ensure-dynamic-classloader
+  "Ensure the context class loader is always a `clojure.lang.DynamicClassLoader`
+  to prevent `ClassNotFoundException`.
+  https://github.com/clj-commons/aleph/issues/603."
+  []
+  (let [thread (Thread/currentThread)
+        context-class-loader (.getContextClassLoader thread)
+        compiler-class-loader (.getClassLoader clojure.lang.Compiler)]
+    (when-not (instance? DynamicClassLoader context-class-loader)
+      (.setContextClassLoader
+        thread (DynamicClassLoader. (or context-class-loader
+                                        compiler-class-loader))))))
```

Comment:
So, my one concern is that `compiler-class-loader` may not have the correct parent classloader set in this code from Riemann, since we need to ensure Netty's classloader is in the chain. I was going to try something like:

```clojure
(let [thread (Thread/currentThread)]
  (when-not (instance? DynamicClassLoader (.getContextClassLoader thread))
    (.setContextClassLoader thread (RT/makeClassLoader))))
```

which seems simpler and (maybe) safer

Reply:
Applying the following change makes the test fail with the initial issue : `ClassNotFoundException`

Reply:
Huh. Interesting